test/Microsoft.ML.OnnxTransformTest/DnnImageFeaturizerTest.cs

```diff
@@ -0,0 +1,204 @@
+            public float[] data_0;
+        }
+        private class TestDataSize
+        {
+            [VectorType(2)]
+            public float[] data_0;
+        }
+        private class TestDataXY
+        {
+            [VectorType(inputSize)]
+            public float[] A;
+        }
+        private class TestDataDifferntType
+        {
+            [VectorType(inputSize)]
+            public string[] data_0;
+        }
+
+        private float[] getSampleArrayData()
+        {
+            var samplevector = new float[inputSize];
+            for (int i = 0; i < inputSize; i++)
+                samplevector[i] = (i / ((float) inputSize));
+            return samplevector;
+        }
+
+        public DnnImageFeaturizerTests(ITestOutputHelper helper) : base(helper)
+        {
+        }
+
+        [ConditionalFact(typeof(Environment), nameof(Environment.Is64BitProcess))]
+        void TestDnnImageFeaturizer()
+        {
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                return;
+            
+
+            var samplevector = getSampleArrayData();
+
+            var dataView = ComponentCreation.CreateDataView(Env,
+                new TestData[] {
+                    new TestData()
+                    {
+                        data_0 = samplevector
+                    },
+                });
+
+            var xyData = new List<TestDataXY> { new TestDataXY() { A = new float[inputSize] } };
+            var stringData = new List<TestDataDifferntType> { new TestDataDifferntType() { data_0 = new string[inputSize] } };
+            var sizeData = new List<TestDataSize> { new TestDataSize() { data_0 = new float[2] } };
+            var appDataBaseDir = Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData);
+            var execDir = Path.Combine(appDataBaseDir, "mlnet-resources");
+            var pipe = new DnnImageFeaturizerEstimator(Env, (m, env, input, output) => m.ResNet18(env, input, output, execDir), "data_0", "output_1");
+
+            var invalidDataWrongNames = ComponentCreation.CreateDataView(Env, xyData);
+            var invalidDataWrongTypes = ComponentCreation.CreateDataView(Env, stringData);
+            var invalidDataWrongVectorSize = ComponentCreation.CreateDataView(Env, sizeData);
+            TestEstimatorCore(pipe, dataView, invalidInput: invalidDataWrongNames);
+            TestEstimatorCore(pipe, dataView, invalidInput: invalidDataWrongTypes);
+            pipe.GetOutputSchema(SchemaShape.Create(invalidDataWrongVectorSize.Schema));
+            try
+            {
+                pipe.Fit(invalidDataWrongVectorSize);
+                Assert.False(true);
+            }
+            catch (ArgumentOutOfRangeException) { }
+            catch (InvalidOperationException) { }
+        }
+
+        [ConditionalFact(typeof(Environment), nameof(Environment.Is64BitProcess))] // x86 fails with "An attempt was made to load a program with an incorrect format."
```

Comment:
We should use a better comment here to mark this test. It should say that `Onnx` is only supported on Windows x64.

Reply:
IMO - it would be best if we created a `Fact` attribute for Onnx tests and marked them as such.

see [https://github.com/dotnet/machinelearning/issues/1220](https://github.com/dotnet/machinelearning/issues/1220)

---
In reply to: [232767475](https://github.com/dotnet/machinelearning/pull/1447#discussion_r232767475) [](ancestors = 232767475)

Reply:
Added a better comment. I generally agree with the Fact assessment, but I don't feel like that belongs as a part of this PR.

---
In reply to: [232768343](https://github.com/dotnet/machinelearning/pull/1447#discussion_r232768343) [](ancestors = 232768343,232767475)